manager/src/main/java/com/redhat/service/bridge/manager/api/EventBridgeManagerExceptionMapper.java

```diff
@@ -1,23 +1,41 @@
 package com.redhat.service.bridge.manager.api;
 
+import java.util.Optional;
+
+import javax.inject.Inject;
 import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.ResponseBuilder;
 import javax.ws.rs.ext.ExceptionMapper;
 import javax.ws.rs.ext.Provider;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.redhat.service.bridge.manager.ErrorsService;
+import com.redhat.service.bridge.manager.api.models.responses.ErrorResponse;
 import com.redhat.service.bridge.manager.exceptions.EventBridgeManagerException;
+import com.redhat.service.bridge.manager.models.Error;
 
 @Provider
 public class EventBridgeManagerExceptionMapper implements ExceptionMapper<EventBridgeManagerException> {
 
+    @Inject
+    ErrorsService errorsService;
+
     private static final Logger LOGGER = LoggerFactory.getLogger(EventBridgeManagerExceptionMapper.class);
 
-    //TODO - Extend this with support for Error codes and useful payload
     @Override
     public Response toResponse(EventBridgeManagerException e) {
         LOGGER.error("Failure", e);
-        return Response.status(e.getStatusCode()).entity(e.getMessage()).build();
+        Optional<Error> error = errorsService.getError(e);
+        ResponseBuilder builder = Response.status(e.getStatusCode());
+        if (error.isPresent()) {
+            ErrorResponse errorResponse = ErrorResponse.from(error.get());
+            errorResponse.setReason(e.getMessage());
+            builder.entity(errorResponse);
+        } else {
+            builder.entity(e.getMessage());
```

Comment:
Since this should not happen, should we `log.warn` ?

Reply:
yes, we should indeed, thanks